addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -1351,17 +1351,21 @@ export class OdooEditor extends EventTarget {
     _setLinkZws() {
         this._resetLinkZws();
         const linkInSelection = getInSelection(this.document, 'a');
-        for (const link of this.editable.querySelectorAll('a')) {
-            if (!isBlock(link) && link.textContent !== '' && !(link.children.length === 1 && link.firstElementChild.nodeName === 'IMG')) {
-                link.prepend(this._createLinkZws('start'));
-                // Only add the ZWS at the end if the link is in selection.
-                if (linkInSelection && link === linkInSelection) {
-                    link.append(this._createLinkZws('end'));
-                }
-                const zwsAfter = this._createLinkZws('after');
-                link.after(zwsAfter);
-                if (!zwsAfter.parentElement || !zwsAfter.parentElement.isContentEditable) {
-                    zwsAfter.remove();
+        const selection = this.document.getSelection();
```

Comment:
> `<p>abc</p><p><a>d[]ef</a></p>` is literally the expected result when pressing the right arrow _twice_.

Uh yes it is, but we only pressed the right arrow _once_ and shouldn't have gotten all the way there.

> I don't understand why there is any tradeoff to be had

If we set the link zws only in the current block, then using the arrows to navigate to another block that starts/ends with a link won't work properly since the link zws won't be in that other block yet. The next commit introduces a solution for that, by setting the link zws in the other block, but that's fairly costly since then we can't just limit the computation to the current block but have to go explore the edges of siblings.

We can talk about this on Thursday, no problem.

Reply:
@dmo-odoo This we still need to discuss this afternoon (cfr the tradeoff and tradeoff-addressing commit).